### ISSUE TITLE
ExtendedProduct should not extend Product from ShopBundle

### ISSUE THREAD

**Issue by TomasLudvik** [shopsys/shopsys#667]
### What is happening
When you add some private property into ShopBundle Product, it fails on error:

```
1) Tests\ShopBundle\Functional\EntityExtension\EntityExtensionTest::testAll
ReflectionException: Property Tests\ShopBundle\Functional\EntityExtension\Model\ExtendedProduct::$distinguishingParameter1 does not exist

/var/www/html/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/RuntimeReflectionService.php:63
/var/www/html/vendor/shopsys/doctrine-orm/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:970
/var/www/html/vendor/shopsys/doctrine-orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:747
/var/www/html/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:316
/var/www/html/vendor/shopsys/doctrine-orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:78
/var/www/html/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:183
/var/www/html/vendor/shopsys/doctrine-orm/lib/Doctrine/ORM/EntityManager.php:283
/var/www/html/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/ObjectManagerDecorator.php:90
/var/www/html/tests/ShopBundle/Functional/EntityExtension/EntityExtensionTest.php:128
/var/www/html/tests/ShopBundle/Functional/EntityExtension/EntityExtensionTest.php:129
/var/www/html/tests/ShopBundle/Functional/EntityExtension/EntityExtensionTest.php:88
```

### Expected result
Adding private property into ShopBundle Product does not fail on error.

**Comment by grossmannmartin** [comment]
Hi @TomasLudvik, 
thank you for submitting an issue, but sadly I wasn't able to reproduce the error. 

Can you please provide a more detailed explanation of the issue? 
I tried to add a private field into `Shopsys\ShopBundle\Model\Product\Product`, generate and run migrations and tests passed ok. 

Are there any other steps to do to reproduce the error?

**Comment by TomasLudvik** [comment]
This was solved by #899
